core/src/test/java/kafka/test/junit/ClusterTestExtensions.java

```diff
@@ -91,9 +91,6 @@ public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContex
         ClusterTemplate clusterTemplateAnnot = context.getRequiredTestMethod().getDeclaredAnnotation(ClusterTemplate.class);
         if (clusterTemplateAnnot != null) {
             processClusterTemplate(context, clusterTemplateAnnot, generatedContexts::add);
-            if (generatedContexts.isEmpty()) {
-                throw new IllegalStateException("ClusterConfig generator method should provide at least one config");
```

Comment:
My previous comment is wrong. This check is used to make sure there is at least one `ClusterConfig`, so we should not remove it. However, It seems to me this check should be moved to `processClusterTemplate` and we should add unit test for it.

Reply:
If we move the check to `processClusterTemplate`, `generatedContexts` also need to expose to `processClusterTemplate`, so I prefer keep this version and revert the change.

Reply:
> try to add unit test for it.

nice. It is ok to file a jira to log it and we can merge this PR first.

Reply:
https://issues.apache.org/jira/browse/KAFKA-16652

@TaiJuWu Please feel free to take over it